lib/system.nim

```diff
@@ -1200,6 +1200,14 @@ const
   hasSharedHeap = defined(boehmgc) or defined(gogc) # don't share heaps; every thread has its own
   taintMode = compileOption("taintmode")
 
+when defined(boehmgc):
```

Comment:
I don't like it in system but I suspect you tried and failed to put it in system/threads.nim.